### PR TITLE
[Helper-Plugin] Migrate useFieldHint to TS

### PR DESCRIPTION
**Pull request by Feranchz**
### What does it do?

Part of #17690 to migrate helper plugin to Typescript

### Related issue(s)/PR(s)

Relates to #17690

### CODE REVIEW

File: packages/core/helper-plugin/src/hooks/tests/useFieldHint.test.tsx [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 
-import { useFieldHint } from '../useFieldHint';
+import { useFieldHint, UseFieldHintProps } from '../useFieldHint';
 
 const messages = { 'message.id': 'response' };
 const knownDescription = { id: 'message.id', defaultMessage: '' };
 
 // eslint-disable-next-line react/prop-types
-export const IntlWrapper = ({ children }) => (
+export const IntlWrapper = ({ children }: { children: React.JSX.Element }) => (
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/hooks/tests/useFieldHint.test.tsx:13]:
```suggestion
const IntlWrapper = ({ children }: { children: React.ReactNode }) => (
```

I think this is the preferred definition for children, what makes you think it should be this? :) 

also does it need to be exported?

File: packages/core/helper-plugin/src/hooks/tests/useFieldHint.test.tsx [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
   <IntlProvider locale="en" messages={messages} textComponent="span">
     {children}
   </IntlProvider>
 );
 
-function setup(args) {
+// @TODO: Remove export any type when setup is typed
+function setup(args: UseFieldHintProps): any {
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/hooks/tests/useFieldHint.test.tsx:20]: Can you demystify the below setup? I don't think it needs to be wrapped in promise and act, this seems like either an issue in our tests or the hook 🤔

File: packages/core/helper-plugin/src/hooks/useFieldHint.tsx [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
+type FieldSchema = {
+  minLength?: number | string;
+  maxLength?: number | string;
+  max?: number | string;
+  min?: number | string;
+};
+export interface UseFieldHintProps {
+  description?: MessageDescriptor & { values?: Record<string, PrimitiveType> };
+  fieldSchema?: FieldSchema;
+  type?: string;
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/hooks/useFieldHint.tsx:15]: This is an interesting one, I suppose this should be something `Attribute.Kind` which is imported from `@strapi/strapi`?

**Review by joshuaellis — CHANGES_REQUESTED** [review]
Some thoughts 🤔

**Review by madhurisandbhor — APPROVED** [review]
LGTM, please remove unnecessary await from tests.

File: packages/core/helper-plugin/src/hooks/tests/useFieldHint.test.tsx [hunk 4]
```diff
@@ -1,40 +1,38 @@
...
 );
 
-function setup(args) {
-  return new Promise((resolve) => {
-    act(() => {
-      resolve(renderHook(() => useFieldHint(args), { wrapper: IntlWrapper }));
-    });
-  });
+function setup(args: UseFieldHintProps) {
+  return renderHook(() => useFieldHint(args), { wrapper: IntlWrapper });
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/hooks/tests/useFieldHint.test.tsx:21]: i'd inline the wrapper, avoids erroneous typing 👍🏼

**Review by madhurisandbhor — APPROVED** [review]

**Pull request merged**